src/materials/ShaderMaterial.js

```diff
@@ -47,6 +47,7 @@ THREE.ShaderMaterial = function ( parameters ) {
 	this.fog = false; // set to use scene fog
 
 	this.lights = false; // set to use scene lights
+	this.userClippingPlanes = false; // set to use user-defined clipping planes
```

Comment:
Hmmm.... are clipping planes material specific?  Or are they global renderer state?  I think this PR sort of confuses the two, or maybe that is on purpose?  I am okay with this but if we can make it one or the other, that would lead to a clearer design (unless this is needed.)